### DISCUSSION TITLE
Why is the value of public key type -1 after loading the SM2 certificate?

### DISCUSSION THREAD

**Discussion by liu-allen** [openssl/openssl#23993] [Q&A] [answered]
Openssl version 3.2.0，
SM2 certificate signed by SM2-WITH-SM3, using function 
X509=PEM_read_bio_X509_AUX (bio, NULL, NULL, NULL); 
EVP_PKEY * pkey=X509_get0_pubkey (x509);
 After loading, the value of pkey ->type is -1。
Does openssl version 3.2.0 not support SM2 certificates signed by SM2-WITH-SM3?

**Comment by t8m** [comment, marked as answer]
This is expected. The -1 value is `EVP_PKEY_KEYMGMT` which means the key is held by a provider. You can check if it has SM2 type by using the EVP_PKEY_is_a() function.
